core/templates/services/math-interactions.service.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for providing helper functions for math interactions.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+const nerdamer = require('nerdamer');
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MathInteractionsService {
+  private warningText = '';
+
+  private cleanErrorMessage(errorMessage: string): string {
```

Comment:
No tests for this function.

Reply:
@nithusha21 This is a private function and is not meant to be called directly. As I understand we only write unit tests for public functions that can be called directly. This is just a helper function and this does indirectly get tested since I am checking for the warning text, which gets generated by this method, after calling the validation function.

Reply:
Ah understood. Thanks!